src/NuGet.Core/NuGet.Protocol/NameSpace/NameSpaceLookupUtility.cs

```diff
@@ -0,0 +1,36 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System.Collections.Generic;
+using System.Linq;
+using NuGet.Configuration;
+
+namespace NuGet.Protocol
+{
+    public static class NameSpaceLookupUtility
+    {
+        public static INameSpaceLookup ConstructNameSpaceLookup(ISettings settings)
+        {
+            INameSpaceLookup nameSpaceLookup = null;
+
+            if (settings == null)
+                return nameSpaceLookup;
+
+            PackageNamespacesConfiguration configuration = PackageNamespacesConfiguration.GetPackageNamespacesConfiguration(settings);
```

Comment:
All of this looks reasonable. The building here can be done in the same object to prevent duplication.

In packages.config there's only 1 of each lookup + configuration, in PackageReference, there are `n` of each, so it's preferrable to have only 1 model for all that.

Reply:
Moved to SearchTree please check if it's good?